inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -758,24 +768,57 @@ void MKLDNNGraph::PullOutputData(BlobMap &out) {
             ext_blob->allocate();
         }
 
-        if (ext_blob->byteSize() != intr_blob.GetSize())
-            THROW_IE_EXCEPTION << "Output blob size is not equal network output size ("
-                               << ext_blob->size() << "!=" << intr_blob.GetSize()/sizeof(float) << ").";
-
         void *ext_blob_ptr = ext_blob->buffer();
         void *intr_blob_ptr = intr_blob.GetData();
 
-        // That is the same memory. No need to copy
-        if (ext_blob_ptr == intr_blob_ptr) continue;
-
         int MB = intr_blob.GetDims()[0];
         int MB_to_process = node->batchToProcess();
         // TODO: Should we support InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_LIMIT???
         if (config.batchLimit)
             MB_to_process = std::min<int>(config.batchLimit, MB_to_process);
         size_t size_to_copy = intr_blob.GetSize() * MB_to_process / MB;
 
-        cpu_memcpy_s(ext_blob_ptr, ext_blob->byteSize(), intr_blob_ptr, size_to_copy);
+        auto srcPrec = MKLDNNMemory::convertToIePrec(intr_blob.GetDataType());
+        auto dstPrec = ext_blob->getTensorDesc().getPrecision();
+        if (dstPrec == srcPrec || (dstPrec == Precision::BOOL && srcPrec == Precision::U8)) {
+            if (ext_blob->byteSize() != intr_blob.GetSize())
+                THROW_IE_EXCEPTION << "Output blob byte size is not equal network output byte size ("
+                                   << ext_blob->byteSize() << "!=" << intr_blob.GetSize() << ").";
+
+            // That is the same memory. No need to copy
+            if (ext_blob_ptr == intr_blob_ptr) continue;
```

Comment:
Why this condition was moved inside `dstPrec == srcPrec` case? Should it be valid for any case?

Reply:
yes, it is. at this moment we change pointer for node on pointer to blob in case:
input if network input precision equals mkldnn graph input node precision
for other if precision equals fp32